main.go

```diff
@@ -43,88 +49,106 @@ func initClientConnection() bpb.BirthdayFunctionsClient {
+	corsConfig.AllowCredentials = true
+	corsConfig.AddAllowHeaders(
+		"x-content-length",
+		"authorization",
+		"cache-control",
+		"x-requested-with",
+		"content-disposition",
+		"content-range",
+		"destination",
+		"fileID",
+	)
+
+	return corsConfig
+}
 
+// CreateBirthday inserts a birthday object if it
+// doesn't exist. if it does, its being overritten
+func (r *Router) CreateBirthday(c *gin.Context) {
 	request := &bpb.CreateBirthdayRequest{
-		PersonalNumber: c.Request.FormValue(personalNumber),
-		Name:           c.Request.FormValue(name),
-		Date:           c.Request.FormValue(date),
+		PersonalNumber: c.Request.FormValue(PersonalNumber),
+		Name:           c.Request.FormValue(Name),
+		Date:           c.Request.FormValue(Date),
 	}
-
 	res, err := r.client.CreateBirthday(c, request)
 	if err != nil {
-		fmt.Println("create birthday error: ", err)
+		log.Fatal("create birthday error: ", err)
 		c.String(400, "create birthday method failed. \nerror: %s", err)
-		os.Exit(3)
 	}
-	c.JSON(201, res)
+	c.JSON(200, res)
 }
 
-// GetBirthday read/get all birthday objects
+// GetBirthday returns a birthday object
 func (r *Router) GetBirthday(c *gin.Context) {
-
-	request := &bpb.GetBirthdayRequest{PersonalNumber: c.Query(personalNumber)}
+	request := &bpb.GetBirthdayRequest{PersonalNumber: c.Query(PersonalNumber)}
 	res, err := r.client.GetBirthday(c, request)
 	if err != nil {
-		fmt.Println("get birthday error: ", err)
+		log.Fatal("get birthday error: ", err)
 		c.String(404, "get birthday method failed. \nerror: %s", err)
-		os.Exit(5)
 	}
 	c.JSON(200, res)
 }
 
-// UpdateBirthday updates a birthday object by personalNumber
-func (r *Router) UpdateBirthday(c *gin.Context) {
-
-	request := &bpb.UpdateBirthdayRequest{PersonalNumber: personalNumber, Name: name, Date: date}
-	res, err := r.client.UpdateBirthday(c, request)
+//GetAllBirthdays returns all birthday objects
+func (r *Router) GetAllBirthdays(c *gin.Context) {
+	request := &bpb.GetAllBirthdaysRequest{}
+	res, err := r.client.GetAllBirthdays(c, request)
 	if err != nil {
-		fmt.Println("update birthday error: ", err)
-		c.String(400, "update birthday method failed. \nerror: %s", err)
-		os.Exit(5)
+		log.Fatal("get all birthdays error.\n", err)
+		c.String(404, "get all birthday method failed. \nerror: %s", err)
 	}
-	c.JSON(201, res)
+	c.JSON(200, res)
 }
 
-// DeleteBirthday deletes a certain birthday object by personal number
+// DeleteBirthday deletes a birthday object by personal number
 func (r *Router) DeleteBirthday(c *gin.Context) {
-
-	request := &bpb.DeleteBirthdayRequest{PersonalNumber: personalNumber}
+	request := &bpb.DeleteBirthdayRequest{PersonalNumber: c.Query(PersonalNumber)}
 	res, err := r.client.DeleteBirthday(c, request)
 	if err != nil {
-		fmt.Println("delete birthday error: ", err)
+		log.Fatal("delete birthday error: ", err)
 		c.String(404, "delete birthday method failed. \nerror: %s", err)
```

Comment:
please use http.(statuscode) from "net/http"

Reply:
and it's not 404

Reply:
done

Reply:
why http.StatusConflict?

Reply:
done